Updates to "Process shift data" shift analyzing

mjweiberg28 opened the pull request:
## Description

1. Justin had an issue with the "process shift data" button where it was showing a 500 error when it was clicked. The reason for this turned out to be that a student had a shift in his uploaded schedule of shifts, but had _no_ shift in the scan data for the program to analyze against. Because there were no shifts to compare to, the program kept searching for that student's name but couldn't find it, continuing until the program gave a 500.
2. The name_present, last_shift additions are to prevent system errors if a student has a shift in the schedule shifts, but not the scan shifts. The method will now look through the remaining shifts for that student's name in any shifts. If the student's name is not found, the method will skip that shift and continue checking other shifts for the other students.
3. The original_n additions are to extend schedule shift's start and end times to match the duration of their shift, not just in the individual shifts of a multiple shift (e.g., 9-11 and 11-1 in succession are now 9-1). This makes for easier readability and less shifts showing in the flagged sheet.

## Size and Type of change

- Small Change - 1 person needs to review this
- Bug fix

## How Has This Been Tested?

- Locally, I tested with the user data that Justin had uploaded. Testing with the same data that had caused the problems allowed me to locate the problem and write up a solution. Testing the same data following these changes resulted in the correct results of "flagged shifts."

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested on multiple browsers (Chrome, Firefox, Safari, IE suite)

mjweiberg28 committed:
Updates to shifts_controller

1. The name_present, last_shift additions are to prevent system errors if a student has a shift in the schedule shifts, but not the scan shifts.
2. The original_n additions are to extend schedule shift's start and end times to match the duration of their shift, not just in the individual shifts of a multiple shift (e.g., 9-11 and 11-1 in succession are now 9-1). This makes for easier readability and less shifts showing in the flagged sheet.

c-schwarze commented on app/shifts_controller.py:
```diff
@@ -216,9 +218,11 @@ def shift_processor(self):
         shift_count = -1  # tracks the current part of the extra shift(s)
         lower_bound = -1
         upper_bound = -1
+        name_present = 0  # set to 0, this means the student's name in hd_shift is present within the scan_shift data
```
The name_present variable could use some work I think. Your comments are fantastic! But it might be better if the variable name and values are more descriptive.

For example, the values 0, 1, and 2 don't mean anything. Maybe use text to represent those values.

c-schwarze commented on app/shifts_controller.py:
```diff
@@ -112,21 +112,23 @@ def flagged_cells(hd_shifts, scan_shifts, hd_row, scan_row, reason, skipped):
-    flag_list.append(
-        dict(zip(flag_key, flagged_cells(hd_shifts, scan_shifts, n, scan_row, cause, skipped))))
+
+    original_n = n  # remembering value of n, so as to set n back to this value after the while loop
```
its probably worth renaming n and original_n to be more descriptive.

c-schwarze reviewed (changes requested):
Looks great overall. Adjusting some of the variable names/values will help bring more clarity

mjweiberg28 committed:
Fixes to PR #22

Addresses comments from PR #22

Variable naming is hopefully clearer now. The original "n" variable for tracking the row of hd_shifts is now hd_row. The "name_present" variable is now "student_name_match" to indicate it tracks whether the student's name exists in both G-sheets and the "original_n" variable is now "placeholder_row" to indicate it exists to save the value of hd_row temporarily.

c-schwarze reviewed (approved):
that looks much better!

mjweiberg28 committed:
Remove prep_scan_shifts method

Adds on to PR #22

I realized this method never actually does anything. It gets called, but the method is not set to any variables and does not return anything, so nothing seems to be done within it that affects the shift analyzing. I tested both before and after removing it and things didn't seem to change at all.
